ast/expressions.go

```diff
@@ -269,7 +269,9 @@ type CaseExpr struct {
 // Format the ExprNode into a Writer.
 func (n *CaseExpr) Format(w io.Writer) {
 	fmt.Fprint(w, "CASE ")
-	n.Value.Format(w)
+	if n.Value != nil {
```

Comment:
Add comment "Because the presence of `case when` syntax, `Value` could be nil and we need check this."

Reply:
great comment